File: app/src/main/java/com/dublikunt/nclient/CopyToClipboardActivity.java
package com.dublikunt.nclient;

import android.content.ClipData;
import android.content.ClipboardManager;
import android.content.Context;
import android.net.Uri;
import android.os.Bundle;
import android.widget.Toast;

import androidx.annotation.NonNull;

import com.dublikunt.nclient.components.activities.GeneralActivity;

public class CopyToClipboardActivity extends GeneralActivity {
    public static void copyTextToClipboard(@NonNull Context context, String text) {
        ClipboardManager clipboard = (ClipboardManager) context.getSystemService(Context.CLIPBOARD_SERVICE);
        ClipData clip = ClipData.newPlainText("text", text);
        if (clipboard != null)
            clipboard.setPrimaryClip(clip);
    }

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        Uri uri = getIntent().getData();
        if (uri != null) {
            copyTextToClipboard(this, uri.toString());
            Toast.makeText(this, R.string.link_copied_to_clipboard, Toast.LENGTH_SHORT).show();
        }
        finish();
    }
}

File: app/src/main/kotlin/com/dublikunt/nclient/CopyToClipboardActivity.kt
package com.dublikunt.nclient

import android.content.ClipData
import android.content.ClipboardManager
import android.content.Context
import android.os.Bundle
import android.widget.Toast

class CopyToClipboardActivity : GeneralActivity() {
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        val uri = intent.data
        if (uri != null) {
            copyTextToClipboard(this, uri.toString())
            Toast.makeText(this, R.string.link_copied_to_clipboard, Toast.LENGTH_SHORT).show()
        }
        finish()
    }

    companion object {
        @JvmStatic
        fun copyTextToClipboard(context: Context, text: String) {
            val clipboard = context.getSystemService(CLIPBOARD_SERVICE) as ClipboardManager
            val clip = ClipData.newPlainText("text", text)
            clipboard?.setPrimaryClip(clip)
        }
    }
}
